plutus-ledger-api/src/PlutusLedgerApi/Common/Versions.hs

```diff
@@ -227,58 +228,42 @@ batch6 =
   where no new builtins are added.  See Note [New builtins/language versions and
   protocol versions]
 -}
-builtinsIntroducedIn :: PlutusLedgerLanguage -> Map.Map MajorProtocolVersion (Set.Set DefaultFun)
+builtinsIntroducedIn :: Map.Map (PlutusLedgerLanguage, MajorProtocolVersion) (Set.Set DefaultFun)
 builtinsIntroducedIn =
-  \case
-    PlutusV1 ->
-      Map.fromList
-      [ (alonzoPV, Set.fromList batch1)
-      , (pv11PV,   Set.fromList (batch2 ++ batch3 ++ batch4 ++ batch5 ++ batch6))
-      ]
-    PlutusV2 ->
-      Map.fromList
-      [ (vasilPV,     Set.fromList (batch1 ++ batch2))
-      , (valentinePV, Set.fromList batch3)
-      , (plominPV,    Set.fromList batch4b)
-      , (pv11PV ,     Set.fromList (batch4a ++ batch5 ++ batch6))
-      ]
-    PlutusV3 ->
-      Map.fromList
-      [ (changPV,  Set.fromList (batch1 ++ batch2 ++ batch3 ++ batch4))
-      , (plominPV, Set.fromList batch5)
-      , (pv11PV,   Set.fromList batch6)
-      ]
+  Map.fromList
+  [ ((PlutusV1, alonzoPV),    Set.fromList batch1)
+  , ((PlutusV1, pv11PV),      Set.fromList (batch2 ++ batch3 ++ batch4 ++ batch5 ++ batch6))
+  , ((PlutusV2, vasilPV),     Set.fromList (batch1 ++ batch2))
+  , ((PlutusV2, valentinePV), Set.fromList batch3)
+  , ((PlutusV2, plominPV),    Set.fromList batch4b)
+  , ((PlutusV2, pv11PV) ,     Set.fromList (batch4a ++ batch5 ++ batch6))
+  , ((PlutusV3, changPV),     Set.fromList (batch1 ++ batch2 ++ batch3 ++ batch4))
+  , ((PlutusV3, plominPV),    Set.fromList batch5)
+  , ((PlutusV3, pv11PV),      Set.fromList batch6)
+  ]
```

Comment:
It's fine, but I kinda liked previous version more, which you can make efficient if you follow [this](https://github.com/IntersectMBO/plutus/pull/7223#discussion_r2229334293).